shared/reducers/team-building.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+// Used in the chat reducer
+import * as I from 'immutable'
+import * as Constants from '../constants/team-building'
+import * as Types from '../constants/types/team-building'
+import * as TeamBuildingGen from '../actions/team-building-gen'
+
+export default function<X, S: I.RecordOf<X & Types.TeamBuildingSubState>>(
+  state: S,
+  action: TeamBuildingGen.Actions
+): S {
+  switch (action.type) {
+    case TeamBuildingGen.resetStore:
+    case TeamBuildingGen.cancelTeamBuilding:
+      return state.merge(Constants.makeSubState())
+    case TeamBuildingGen.addUsersToTeamSoFar:
+      return state.update('teamBuildingTeamSoFar', teamSoFar => teamSoFar.merge(action.payload.users))
+    case TeamBuildingGen.removeUsersFromTeamSoFar: {
+      const setToRemove = I.Set(action.payload.users)
+      return state.update('teamBuildingTeamSoFar', teamSoFar => teamSoFar.filter(u => !setToRemove.has(u.id)))
+    }
+    case TeamBuildingGen.searchResultsLoaded: {
+      const {query, service, users} = action.payload
+      return state.update('teamBuildingSearchResults', searchResults =>
+        searchResults.set(I.List([query, service]), users)
```

Comment:
is it worth doing a tuple as a key, vs just having it be a nested map? then you can just do
`searchResults.getIn([query, service])`

Reply:
I like it! It makes the search result counts much simpler